### ISSUE TITLE
Yandex gradient boosting tweaks - incresing tree complexity

### ISSUE THREAD

**Issue by gugatr0n1c** [dmlc/xgboost#1146]
Hi,

I was on machinelearning conference here:
http://www.mlprague.com/

and Micheal Levin from Yandex shrared some tweaks they are using in MatrixNet (their gradient boosting implementation).

There were severals tweaks:

Oblivious trees:
http://wan.poly.edu/KDD2012/forms/workshop/ADKDD12/doc/a3.pdf
- for speed up and more generalization (but not sure what this does with accuracy)
- anyone tried this?

Increasing tree complexity:
- use different tree complexity during the learning
- for first iterations use ie. max_depth = 4, min_child_weight = 50, for next iterations use max_depth = 7, min_child_weight = 10, finally use maxx_depth = 11, min_child_weight = 1
- actually this can help speed up (lower complexity for firts iterations can save time) and if generalization works well it can help accuracy as well

In this issue I proposing to ability to set all tree parameters as vector for each boosting round (there was similar issue for learning_rate) - then we can increase tree complexity during process.

But would be good first investigate possible bug here:
https://github.com/dmlc/xgboost/issues/1054

Thx for consideration

**Comment by Far0n** [comment]
Regarding: _Increasing tree complexity_:  Not the most elegant way, but you can do that already with training continuation:

```
xgb_params['depth'] = 4
gbdt = xgb.train(xgb_params, dtrain, 50)
xgb_params['depth'] = 7
gbdt = xgb.train(xgb_params, dtrain, 50, xgb_model=gbdt)
```
